tests/system/utils.py

```diff
@@ -39,3 +40,122 @@ def get_resource(resource):
+       fault domain information
+    """
+    agent_domains = {}
+
+    # Populate agent_domains with the ID and the corresponding domain for each slave
+    master = mesos.get_master()
+    for slave in master.slaves():
+        agent_domains[slave['id']] = FaultDomain(slave._short_state.get('domain', None))
+    return agent_domains
+
+
+def get_all_cluster_regions():
+    """Returns a dictionary with all the regions and their zones in the cluster
+    """
+    domain_regions = {}
+
+    # Populate all the domain zones and regions found in the cluster
+    for domain in get_cluster_agent_domains().values():
+        if domain.region not in domain_regions:
+            domain_regions[domain.region] = []
+        if domain.zone not in domain_regions[domain.region]:
+            domain_regions[domain.region].append(domain.zone)
+
+    # Return dictionary
+    return domain_regions
+
+
+def get_biggest_cluster_region():
+    """Returns a tuple with the name of the region with the most zones in the
+       cluster and a list with the actual zones in it
+    """
+    biggest_region = None
+    biggest_region_zones = []
+
+    for (region, zones) in get_all_cluster_regions().items():
+        if len(zones) > len(biggest_region_zones):
+            biggest_region_zones = zones
+            biggest_region = region
+
+    return (biggest_region, biggest_region_zones)
+
+
+def get_app_domains(app):
+    """Returns a list of all te fault domains used by the tasks of the specified app
+    """
+    tasks = app.get('tasks', [])
+    agent_domains = get_cluster_agent_domains()
+
+    assert len(tasks) > 0, "App %s did not launch any tasks on mesos" % (app['id'],)
+
+    # Collect the FaultDomain objects from all the agents where the tasks are running
+    domains = set()
+    for task in tasks:
+        domains.append(agent_domains[task['slaveId']])
+
+    return domains
+
+
+def get_used_regions_and_zones(domains):
+    """Returns tuple with the sets of the used regions and zones from the given list
+       of FaultDomain object instances (ex. obtained by get_app_domains())"""
+    used_regions = set()
+    used_zones = set()
+
+    for domain in domains:
+        used_regions.add(domain.region)
+        used_zones.add(domain.zone)
+
+    return (used_regions, used_zones)
+
+
+def count_agents_in_faultdomains(regions=None, zones=None):
+    """Return the number of agents that belong on the specified region and/or zone
+    """
+    counter = 0
+
+    # Make sure to always operate on iterable
+    if type(regions) is str:
+        regions = [regions]
+    if type(zones) is str:
+        zones = [zones]
+
+    # Increment counter for the agents that pass the checks
+    for domain in get_cluster_agent_domains().values():
+        if regions is not None and domain.region not in regions:
+            continue
+        if zones is not None and domain.zone not in zones:
+            continue
+        counter += 1
+
+    return counter
```

Comment:
If I understand this correctly you want something like

```python
len([d for d in get_cluster_agent_domains().values()
     if d.region in regions and d.zone in zones])
```

Reply:
Almost.. this function has multiple signatures and this one-liner is not handling it:
* `count_agents_in_faultdomains(regions=[...])` count agents in these region(s)
* `count_agents_in_faultdomains(zones=[...])` count agents in these zone(s)
* `count_agents_in_faultdomains(regions=[], zones=[])` count agents in these regions AND zones.

Reply:
Hm, it feels this is actually a filter. The count should be done afterwards.

Reply:
Whatever you feel more comfortable with. :man_shrugging: 
I personally don't like creating an entire array just to count it's items, but if you think it makes the code cleaner, I could have a look.

Reply:
Yeah, seems like we should have a generic count function which receives a predicate, right?

```
def count(arr, predicate):
    return sum(1 for x in arr if predicate(x))
```

Reply:
That sounds like a good idea. Let me try it.